src/MessageWriter.ts

```diff
@@ -344,27 +363,42 @@ function time(value: unknown, _defaultValue: DefaultValue, writer: CdrWriter): v
   writer.uint32(timeObj.nsec ?? timeObj.nanosec ?? 0);
 }
 
-function boolArray(value: unknown, defaultValue: DefaultValue, writer: CdrWriter): void {
+function boolArray(
+  value: unknown,
+  defaultValue: DefaultValue,
+  writer: CdrWriter,
+  arrayLength?: number,
+): void {
   if (Array.isArray(value)) {
     const array = new Int8Array(value);
     writer.int8Array(array);
   } else {
-    writer.int8Array((defaultValue ?? []) as number[]);
+    writer.int8Array((defaultValue ?? new Array(arrayLength).fill(0)) as number[]);
```

Comment:
Just wondering, it seems like these CdrWriter `writer.*array` methods have a `writeLength` parameter... why is the MessageWriter not using that but instead doing its own `sequenceLength` call on L213?